sstables/row.hh

```diff
@@ -448,7 +449,7 @@ private:
                 break;
             }
         case state::CELL_VALUE_BYTES:
-            if (read_bytes(data, _u32, _val) != read_status::ready) {
+            if ((_deleted ? read_bytes(data, _u32, _val) : read_bytes(data, _u32, _val_fragmented)) != read_status::ready) {
```

Comment:
This is somewhat messy. How do we keep track when we parse into _val and when into _patse_fragmented?

Will it be cleaner if we parse into _val_fragmented in the _deleted case too?

Reply:
It will be a bit more complex but easier to read, so I'll change it to parse into _val_fragmented first and then, if possible, into _val.